### PR TITLE
XDR mirroring feature

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25898

## Description
Adding XDR incident mirroring:

- [x] make sure which fields return via the API and to which fields should they be mapped.
- [x] in fetch incidents -> list all incidents -> get extra data for all incident ID
- [x] create get-remote-data 
- [x] create update-remote-system
- [x] create get-mapping fields
- [x] create mapping from XDR severity to Demisto severity (existing script xdr sync already does it)
- [x] map xdr status to incident status and upon incident closure in demisto ask to provide reason.
- [x] check incident handling v3 error
- [x] deprecate xdr sync script from version 6.0.0
- [x] change xdrstatus to be a single select field with "under_investigation" and "new" as options.
- [x] Add checkbox param to sync incident owners (uses getUserByUsername and getUserByMail methods)
- [x] Modify alert fields to be without "_" - make sure all significant alerts fields are mapped to the grid layout.
- [x] Add sync parameter for dbotMirrorDirection (incoming, outgoing, Both, None(default)), needs to have clear tooltip
- [x] Add mirroring to dbotMirrorInstance.
- [x] Add error entry in cases of failed incoming and outgoing commands
   - [x] force error experiment - empty error is being printed, checking cause
- [x] Add last sync time to layout
- [x] Create new field - sync error

A continuation of the previous PR: https://github.com/demisto/content/pull/8127 that got corrupted.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 1]
```diff
@@ -16,6 +19,46 @@
...
+}
+
+XSOAR_RESOLVED_STATUS_TO_XDR = {
+    'Resolved': 'resolved_threat_handled',
+    'Other': 'resloved_other',
+    'Duplicate': 'resolved_duplicate_incident',
+    'False Positive': 'resolved_false_positive'
+}
+
+MIRROR_DIRECTION = {
```
Comment (ronykoz) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:56]: let's add that to common

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 2]
```diff
@@ -1624,6 +1667,229 @@ def endpoint_scan_command(client, args):
...
+    if ('assigned_user_mail' in update_args and update_args.get('assigned_user_mail') in ['None', 'null', '', None]) \
+            or ('assigned_user_pretty_name' in update_args
+                and update_args.get('assigned_user_pretty_name') in ['None', 'null', '', None]):
+        update_args['unassign_user'] = 'true'
+        update_args['assigned_user_mail'] = None
+        update_args['assigned_user_pretty_name'] = None
+
+
+def handle_outgoing_issue_closure(update_args, inc_status):
+    if inc_status == 2:
```
Comment (ronykoz) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:1852]: we have a const for that in common

**Review by ronykoz — APPROVED** [review]
Looking good, @anara123 can we go ahead and merge this?

**Comment by anara123** [comment]
@ronykoz @roysagi you can go ahead and merge

**Comment by roysagi** [comment]
@kirbles19 - if you could please check the following files:
1) The [RN](https://github.com/demisto/content/blob/xdr_mirroring_feature/Packs/CortexXDR/ReleaseNotes/2_4_0.md)
2) [XDR integration yml](https://github.com/demisto/content/blob/xdr_mirroring_feature/Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.yml)
3) [The incident handling V3 playbook yml](https://github.com/demisto/content/blob/xdr_mirroring_feature/Packs/CortexXDR/Playbooks/Cortex_XDR_incident_handling_v3.yml) - though should be noted it's the same as the v2
4)

File: Packs/Base/pack_metadata.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
     "name": "Base",
     "description": "The base pack for Cortex XSOAR.",
     "support": "xsoar",
-    "currentVersion": "1.1.13",
+    "currentVersion": "1.1.17",
```
[the remark this answers is not in the document]
Comment (roysagi) [reply Packs/Base/pack_metadata.json]: it was, updated it and made validation pass

**Comment by yaron-libman** [comment]
Reviewed and approved.

**Pull request merged**